ouroboros-consensus/test-consensus/Test/Consensus/HardFork/Combinator/A.hs

```
@@ -280,69 +282,10 @@ instance LedgerSupportsMempool BlockA where
-              -- /before/ the start of the next era.
-              --
-              -- Thus, we schedule the next era to begin with the first
-              -- upcoming epoch that starts /after/ we're guaranteed to see
-              -- both the aforementioned @k@ additional blocks and also a
-              -- further @safeFromTipA k@ slots after the last of those.
-
-              -- The last slot that must be in the current era
-              firstPossibleLastSlotThisEra =
-                  History.addSlots (stabilityWindowA k + safeFromTipA k) sno
-
-              -- The 'EpochNo' corresponding to 'firstPossibleLastSlotThisEra'
-              --
-              -- NOTE: We cannot use 'EpochInfo' for this. The 'EpochInfo'
-              -- only allows us to look ahead as far as the safe zone allows,
-              -- but here we are, almost by definition, looking further ahead
-              -- than that (stability window + safe zone).
-              --
-              -- (Once we construct an 'EpochInfo' with the known transition, it
-              -- would allow us to look this far ahead, but we cannot use it
-              -- to /determine/ the transition point.)
-              lastEpochThisEra =
-                  History.addEpochs
-                    (History.countSlots
-                       firstPossibleLastSlotThisEra firstSlotTipEpoch
-                     `div` unEpochSize epochSizeTipEpoch)
-                    tipEpoch
-
-              -- The first epoch that may be in the next era (recall: eras are
-              -- epoch-aligned)
-              firstEpochNextEra = succ lastEpochThisEra
-
-          return $ Ticked sno $ st {
-              lgrA_transition = Just (sno, firstEpochNextEra)
-            }
-    where
-      k = lcfgA_k ledgerConfig
-
-      runEI :: (EpochInfo Identity -> a -> Identity b) -> a -> b
-      runEI f x = runIdentity $ f ei x
+          return $ Ticked sno $ st { lgrA_transition = Just sno }
```

While you're in here, I was tempted earlier to use the `Latest` monoid if only for indicativeness. Just a thought.

---

I guess I should throw an error if there is more than one. But it doesn't really matter, we can polish that if/when we need to :)